__tests__/List.ts

```diff
@@ -484,6 +484,24 @@ describe('List', () => {
     expect(r.toArray()).toEqual(['b', 'd', 'f']);
   });
 
+  it('filters values based on type', () => {
+    class A {}
+    class B extends A {
+      b(): void {}
+    }
+    class C extends A {
+      c(): void {}
+    }
+    let l1 = List<A>([ new B(), new C(), new B(), new C() ]);
+    // Note: TypeScript requires a defined type guard to reduce the type. Since
+    // arrow functions cannot define return types, we must use a normal function.
+    let l2: List<C> = l1.filter(function (v): v is C {
```

Comment:
Not sure that comment is correct. Arrow functions do support return types. 

http://www.typescriptlang.org/play/#src=class%20A%20%7B%20%7D%0D%0A%5Bnew%20A()%5D.filter((v)%3A%20v%20is%20A%20%3D%3E%20v%20instanceof%20A)

Or is this about something else?